synapse/handlers/presence.py

```diff
@@ -958,17 +1080,33 @@ async def bump_presence_active_time(self, user: UserID) -> None:
 
         bump_active_time_counter.inc()
 
-        prev_state = await self.current_state_for_user(user_id)
+        now = self.clock.time_msec()
 
-        new_fields: Dict[str, Any] = {"last_active_ts": self.clock.time_msec()}
-        if prev_state.state == PresenceState.UNAVAILABLE:
-            new_fields["state"] = PresenceState.ONLINE
+        # Update the device information & mark the device as online if it was
+        # unavailable.
```

Comment:
I feel like this is a "what" comment rather than a "why" comment? Why can't we go from offline -> online? Presumably because we need to have a sync stream to transition out of offline, which we clearly don't have?

Reply:
That's fair, I think the code is just a bit complicated and was attempting to put a succinct description of what's going on.

Reply:
I think this is pretty simplified now, but we still do call `_combine_presence_state` in a few spots.